lib/ansible/modules/cloud/amazon/ec2_ami.py

```diff
@@ -656,7 +656,7 @@ def main():
         virtualization_type=dict(default='hvm'),
         root_device_name=dict(),
         delete_snapshot=dict(default=False, type='bool'),
-        name=dict(),
+        name=dict(default=''),
```

Comment:
I would like to see this fixed differently on devel because now the user gets a traceback if they omit name and image id and the message does not indicate image_id a possibility for state=present. I've opened https://github.com/ansible/ansible/pull/38972 and if that is merged will cherry pick the commit onto this pull request.